src/states/node.rs

```diff
@@ -997,7 +999,10 @@ impl Node {
                              peer_id: PeerId,
                              dst_id: PeerId)
                              -> Result<(), RoutingError> {
-        self.peer_mgr.tunnelling_to(&dst_id);
+        if !self.peer_mgr.tunnelling_to(&dst_id) {
+            debug!("{:?} Received TunnelSuccess for a peer we are already connected to: {:?}", self, dst_id);
```

Comment:
We should send a `TunnelDisconnect` in this case, to indicate to the tunnelling node that we don't need the tunnel connection.